modules/social_features/social_group/social_group.module

```diff
@@ -2100,3 +2100,32 @@ function social_group_social_group_action_social_group_send_email_action_finish(
     'plural' => 'Your email has not been sent to @count selected members successfully',
   ];
 }
+
+/**
+ * Implements hook_preprocess_field().
+ */
+function social_group_preprocess_field(&$variables) {
+  $formatter = $variables['element']['#formatter'];
+  if (in_array($formatter, ['address_plain', 'address_default'])) {
+    $entity = $variables['element']['#object'];
+    if ($entity && $entity instanceof GroupInterface) {
+      $social_group_settings = \Drupal::config('social_group.settings');
+      $address_visibility_settings = $social_group_settings->get('address_visibility_settings');
+      if (isset($address_visibility_settings['street_code_private']) && !empty($address_visibility_settings['street_code_private'])) {
+        if (!$entity->getMember(\Drupal::currentUser())) {
+          switch ($formatter) {
+            case 'address_plain':
+              $variables['items'][0]['content']['#address_line1'] = NULL;
```

Comment:
@denysyukm Maybe it would be good to add some check if this variable exists? Just in case if something will be changed in future.

Reply:
Added